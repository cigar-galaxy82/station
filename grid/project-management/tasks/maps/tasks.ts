/* eslint-disable jest/no-export */

import { SuperfaceTest } from '@superfaceai/testing';
import { nockConfig } from '../../../test-config';

export const createTaskTest = (
  provider: string,
  params: {
    title: string;
    project: string;
    parent?: string;
    assignee?: string;
  }
): void => {
  describe(`project-management/tasks/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest(
        {
          profile: 'project-management/tasks',
          provider,
        },
        nockConfig
      );
    });

    describe('CreateTask', () => {
      describe('when all inputs are correct', () => {
        it('should create a task', async () => {
          const result = await superface.run({
            useCase: 'CreateTask',
            input: params,
          });
          expect(result).toMatchSnapshot();
        });
      });
    });
  });
};
