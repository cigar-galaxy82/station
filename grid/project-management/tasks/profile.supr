"""
Manage tasks in project management systems.
"""
name = "project-management/tasks"
version = "1.0.0"

"""
Create a new task.
"""
usecase CreateTask {
	input {

    """
    projects
    An array of IDs of projects this task belongs to.
    """
    projects! [string]!

		"""
		label
		Textual label of the task.
		"""
		label! string!
		
		"""
		description
		Free-form textual description of the task.
		"""
		description string
	
    """
    parent
    ID of the parent task (if applicable).
    """
    parent string

    """
    assignee
    ID of the user this task should be assigned to.
    """
    assignee string
	}

	result {
    "ID of the just created task."
    id! string!

    "Project IDs it was successfully added to."
    projects! [string]!
  }

	error {
		"Human-readable error message"
		message! string!
		"Human-readable details about the error"
		description! string!
	}

  example success {
    input {
      projects = ["46ebca75-68af-42a1-916d-b228c2082d4f"]
      label = "Get milk"
      description = "We ran out of milk :("
    }

    result {
      id = "ffa18e32-45e6-4b2b-97fa-b8509b3a8c76"
      projects = ["46ebca75-68af-42a1-916d-b228c2082d4f"]
    }
  }
}