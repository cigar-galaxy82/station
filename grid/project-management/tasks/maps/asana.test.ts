import { createTaskTest } from './tasks';

createTaskTest('asana', {
  label: 'Hello, World!',
  projects: ['1202159846567214'],
  parent: '1202225310829823',
  assignee: '1202159845987230',
});
