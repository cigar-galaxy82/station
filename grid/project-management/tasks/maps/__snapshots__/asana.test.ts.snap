// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-management/tasks/asana CreateTask when all inputs are correct should create a task 1`] = `
Ok {
  "value": Object {
    "assignee": "1203384186605863",
    "description": "Description of test task",
    "id": "1203399380711060",
    "parent": null,
    "projects": Array [
      "1203384186840261",
    ],
    "title": "Hello, World!",
    "url": "https://app.asana.com/0/1203384186840261/1203399380711060",
  },
}
`;

exports[`project-management/tasks/asana DeleteTask when all inputs are correct should delete a task 1`] = `
Ok {
  "value": Object {},
}
`;

exports[`project-management/tasks/asana ReadAllTasks when all inputs are correct should read all tasks for specified project 1`] = `
Ok {
  "value": Array [
    Object {
      "id": "1203399327658817",
      "title": "Test 3",
    },
    Object {
      "id": "1203399260234149",
      "title": "Test 2",
    },
    Object {
      "id": "1203399327333641",
      "title": "Test 1",
    },
  ],
}
`;

exports[`project-management/tasks/asana ReadTask when all inputs are correct should read task specified by id 1`] = `
Ok {
  "value": Object {
    "assignee": "1203384186605863",
    "description": "Task description",
    "id": "1203399397001762",
    "parent": null,
    "projects": Array [
      "1203384186840261",
    ],
    "title": "Test",
    "url": "https://app.asana.com/0/1203384186840261/1203399397001762",
  },
}
`;

exports[`project-management/tasks/asana UpdateTask when all inputs are correct should update a task 1`] = `
Ok {
  "value": Object {
    "assignee": "1203384186605863",
    "description": "Description of test task",
    "id": "1203399327963727",
    "parent": null,
    "projects": Array [
      "1203384186840261",
    ],
    "title": "Test Task",
    "url": "https://app.asana.com/0/1203384186840261/1203399327963727",
  },
}
`;
